### PR TITLE
Layout component tweaks

### PR DESCRIPTION
**Pull request by mperrotti**
Creating a PR so I can have use the beta release to see what side effects these changes have in our UIs.

#### Related issues
Fixes https://meetup.atlassian.net/browse/SDS-244

#### Description
We were seeing issues, especially with Section, when using in mup-web to build Event Home.

### CODE REVIEW

**Comment by mperrotti** [comment]
Tested these changes on MUP home, the updated registration flow, and Meetup Pro. Nothing appeared broken, but we may see some unwanted bottom borders since `<Section>` now has a bottom border and bottom padding by default (similar to how `<Section padded>` worked).

Ready for review and merge!

File: src/layout/section.story.jsx [hunk 1]
```diff
@@ -18,47 +20,104 @@ storiesOf('Section', module)
...
+							<div className='chunk'>
+								<h2 className='text--display666'>These sections are padded on all sides</h2>
+							</div>
+						</Section>
+						<Section>
+							<div className='chunk'>
+								<h2 className='text--display666'>These sections are padded on all sides</h2>
+							</div>
+						</Section>
+						<Section> {/* canBeLastChild='large' className='section--notAlwaysLast'*/}
```
Comment (akdetrick) [inline src/layout/section.story.jsx]: These props don't look like they should be commented out.

File: assets/scss/components/_section.scss [hunk 2]
```diff
@@ -11,25 +11,25 @@ $sectionSpaceMultiple: 2;
...
-.section--padded {
-	position: relative;
-	border-top: 0;
-	&:before {
-		content: "";
-		background: $C_border;
-		height: 1px;
-		bottom: 0;
-		position: absolute;
+@include _bpModifier(section, canBeLastChild) {
```
Comment (akdetrick) [inline assets/scss/components/_section.scss]: I'm a little confused about this modifier; in responsive layouts, the position of the `section` in the DOM won't change. 

If the position of the element in the DOM does change via the `render` method in a component, then adding or removing the `section--noBorder` should be sufficient to handle adding or removing the section border.

Comment (mperrotti) [inline assets/scss/components/_section.scss]: We've actually talked about these styles - there's a lot of simplification I need to include.

Comment (akdetrick) [inline assets/scss/components/_section.scss]: Are the props commented out in the story still relevant? Going to take a look in a browser to get a better understanding again.

Comment (mperrotti) [inline assets/scss/components/_section.scss]: No, they're not. I'm going with a different interface

**Pull request merged**